File: Code/Asteroid_Trajectory_Selection/TWO_BODY_SIM.m
function [X, T] = TWO_BODY_SIM(t_final, x_0, MYU_SUN)
    options = odeset('RelTol', 1e-4, 'AbsTol', 1e-4);
    [T, X] = ode113(@(t, x) dx_TBS_dt(t, x, MYU_SUN), [0, t_final], x_0, options);
end

function x_dot = dx_TBS_dt(~, x, MYU_SUN)
    r_vec = x(1:3)';
    r_mag = norm(r_vec);

    r_ddot = - (MYU_SUN * r_vec ./ (r_mag.^3));
    x_dot = [x(4), x(5), x(6), r_ddot(1), r_ddot(2), r_ddot(3)]';
end

File: Python/asteroid_trajectory_selection/two_body_sim.py
"""Two-body simulation using scipy's solve_ivp (replaces MATLAB ode113)."""

import numpy as np
from scipy.integrate import solve_ivp


def two_body_sim(t_final, x_0, mu_sun):
    """Propagate a two-body trajectory from t=0 to t_final."""

    def dx_dt(t, x):
        r_vec = x[0:3]
        r_mag = np.linalg.norm(r_vec)
        r_ddot = -mu_sun * r_vec / (r_mag ** 3)
        return np.array([x[3], x[4], x[5], r_ddot[0], r_ddot[1], r_ddot[2]])

    sol = solve_ivp(
        dx_dt,
        [0, t_final],
        x_0,
        method='DOP853',
        rtol=1e-4,
        atol=1e-4,
        dense_output=True,
    )

    T = sol.t
    X = sol.y.T  # shape (N, 6), rows are time steps

    return X, T
